types/types.go

```diff
@@ -43,77 +43,74 @@ type Endpoint struct {
-	// container networking mode. if set to `host` the host networking will be used for this node, else bridged network
-	NetworkMode          string
-	MgmtNet              string // name of the docker network this node is connected to with its first interface
-	MgmtIntf             string // can be used to be rendered by the default node template 
-	MgmtIPv4Address      string
-	MgmtIPv4PrefixLength int
-	MgmtIPv6Address      string
-	MgmtIPv6PrefixLength int
-	MacAddress           string
-	ContainerID          string
-	TLSCert              string
-	TLSKey               string
-	TLSAnchor            string
-	NSPath               string   // network namespace path for this node
-	Publish              []string // list of ports to publish with mysocketctl
-	ExtraHosts           []string // Extra /etc/hosts entries for all nodes
-	// container labels
-	Labels map[string]string
-	// Slice of pointers to local endpoints
-	Endpoints []Endpoint
+	ShortName            string            `json:"shortname,omitempty"` // name of the Node inside topology YAML
+	LongName             string            `json:"longname,omitempty"`  // containerlab-prefixed unique container name
+	Fqdn                 string            `json:"fqdn,omitempty"`
+	LabDir               string            `json:"labdir,omitempty"` // LabDir is a directory related to the node, it contains config items and/or other persistent state
+	Index                int               `json:"index,omitempty"`
+	Group                string            `json:"group,omitempty"`
+	Kind                 string            `json:"kind,omitempty"`
+	StartupConfig        string            `json:"startupconfig,omitempty"`        // path to config template file that is used for startup config generation
+	StartupDelay         uint              `json:"startupdelay,omitempty"`         // optional delay (in seconds) to wait before creating this node
+	EnforceStartupConfig bool              `json:"enforcestartupconfig,omitempty"` // when set to true will enforce the use of startup-config, even when config is present in the lab directory
+	ResStartupConfig     string            `json:"resstartupconfig,omitempty"`     // path to config file that is actually mounted to the container and is a result of templation
+	Config               *ConfigDispatcher `json:"-"`                              // Do not marshal into JSON - internals
+	ResConfig            string            `json:"resconfig,omitempty"`            // path to config file that is actually mounted to the container and is a result of templation
+	NodeType             string            `json:"nodetype,omitempty"`
+	Position             string            `json:"position,omitempty"`
+	License              string            `json:"license,omitempty"`
+	Image                string            `json:"image,omitempty"`
+	Sysctls              map[string]string `json:"-"` // Do not marshal into JSON - internals
```

Comment:
It is an option, but let's consider the following:
1. In Graphite, `topology-data.json` is served over HTTP w/o authentication, so it is really important to protect the user from unintentionally disclosing sensitive data about **production** network that may be replicated in clab like private TLS keys.
2. The option with omitting fields in the template assumes there is good tracking between structs that can exported and the template, and all sensitive fields are always omitted. But I don't think this will be the case long term – at some moment, more sensitive fields will be addd to NodeConfig or Clab, but not to the template. This is less likely to happen when you have those tags right there in the struct – as a reminder and a pattern to follow.
3. Omitting fields in the template makes it more complex, and that stuff has a good chance to be thrown away during any edits.
4. If we stick with using tags to hide fields only for a limited set of sensitive data, anyone who REALLY needs them can export those explicitly with little effort: `{{ .NodeConfigs.node.TLSKey }}`

Maybe we could do the following: 
* limit tags `json:-` only to the fields in `NodeConfig` that are 100% sensitive. Basically, that is only `TLSKey`
* In `auto.tmpl` instead of marshaling the whole object, just list the subset of the fields that have high value for the default export – thus, no bulky and low value fields would be exported by default.

Reply:
I would be ok with tags if that was true

> If we stick with using tags to hide fields only for a limited set of sensitive data, anyone who REALLY needs them can export those explicitly with little effort: {{ .NodeConfigs.node.TLSKey }}

but I think it is not that easy, because now if you need to export those hidden fields you can't rely in json marshalling anymore, you would need to explicitly mention every other field. Or am I wrong here?

As a template designer I would be ok with keeping track of omitted fields, should we end up taking this route. This is kind of common sense to make sure that changes you made in one place are propagated to the other one. We can/should have a test case after all that checks that a certain deployed topology results in a certain topology-data.json present.

With that said, I am ok with having a single TLSKey field omitted and default template to list explicitly the fields that are exported.

Reply:
See the new template and re-worked tags for NodeConfig

Reply:
There are two fields in total I tagged to omit: (1) TLSKey, (2) Endpoints. The second one has to be omitted via a tag, as it has cyclic references that can't be marshaled (marshal will through an error).

Here is gotemplate code to force `TLSKey` to be exported explicitly, with all other keys being marshaled. 
````
  "nodes": { {{$i:=0}}{{range $n, $c := .NodeConfigs}}{{if $i}},{{end}}{{ $k := dict "tls-key" $c.TLSKey }}
    "{{$n}}": {{ $cj := $c | data.ToJSON | data.JSON }} {{ $dst := coll.Merge $cj $k }}{{ ToJSONPretty $dst "    " "  " }}{{$i = add $i 1}}{{end}}
  },
````

For convenience, I've included this example with another export template `full.tmpl`.

Note, I'm not using `gomplate` to pretty print JSON via `data.ToJSONPretty`, but instead use locally defined in `export.go` `ToJSONPretty`. Turns out, `gomplate` has a bug that does not marshal `Links` struct from `CLab` in accordance with JSON spec – `jq` refuses it due to keys being `int`. Local `ToJSONPretty` converts those to strings, and also does a better job on indentation.

Please let me know if we can settle on this approach.